droidmediabuffer.cpp

```diff
@@ -138,31 +159,83 @@ DroidMediaBuffer *droid_media_buffer_create_from_raw_data(uint32_t w, uint32_t h
     return NULL;
   }
 
-  dstStride = buffer->getStride();
+  linesY = h;
+  dstStrideY = buffer->getStride();
+
+  switch (format) {
+  case HAL_PIXEL_FORMAT_RGBA_8888:
+  case HAL_PIXEL_FORMAT_RGBX_8888:
+  case HAL_PIXEL_FORMAT_BGRA_8888:
+    lineWidthY = 4 * w;
```

Comment:
I've been testing this with RGBA, and it seems that the dstStrideY also needs multiplying by the bytes per pixel.

Reply:
Will fix that, but does the srideY (passed from gst-droid) have to be multiplied too?

Reply:
GST_VIDEO_INFO_COMP_STRIDE seems to already return the stride in bytes there.